Tests/Algorithm/Framework/Alphas/CommonAlphaModelTests.cs

```diff
@@ -0,0 +1,265 @@
+                    var low = last / 1.005m;
+
+                    var packets = _algorithm.Securities.Select(kvp =>
+                    {
+                        var security = kvp.Value;
+                        var tradeBar = new TradeBar(time, security.Symbol, last, high, low, last, 1000);
+                        var configuration = security.Subscriptions.FirstOrDefault();
+                        return new DataFeedPacket(security, configuration, new List<BaseData> { tradeBar });
+                    });
+
+                    return TimeSlice.Create(time, TimeZones.NewYork, new CashBook(), packets.ToList(), changes).Slice;
+                });
+        }
+
+        private bool TryCreateAlphaModel(Language language, out IAlphaModel model)
+        {
+            model = default(IAlphaModel);
+
+            try
+            {
+                switch (language)
+                {
+                    case Language.CSharp:
+                        model = CreateCSharpAlphaModel();
+                        return true;
+                    case Language.Python:
+                        model = CreatePythonAlphaModel();
+                        return true;
+                    default:
+                        return false;
+                }
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Implements a History provider that always return a IEnumerable of Slice with prices following a sine function
+        /// </summary>
+        internal class SineHistoryProvider : IHistoryProvider
+        {
+            private QCAlgorithmFramework _algorithm;
+
+            public int DataPointCount => 0;
+
+            public SineHistoryProvider(QCAlgorithmFramework algorithm)
+            {
+                _algorithm = algorithm;
+            }
+
+            public void Initialize(AlgorithmNodePacket job, IDataProvider dataProvider, IDataCacheProvider dataCacheProvider, IMapFileProvider mapFileProvider, IFactorFileProvider factorFileProvider, Action<int> statusUpdate)
+            {
+            }
+
+            public IEnumerable<Slice> GetHistory(IEnumerable<Data.HistoryRequest> requests, DateTimeZone sliceTimeZone)
+            {
+                foreach (var request in requests)
+                {
+                    Security security;
+                    if (!_algorithm.Securities.TryGetValue(request.Symbol, out security))
+                    {
+                        continue;
+                    }
+
+                    var utc = request.StartTimeUtc;
+                    var end = request.EndTimeUtc;
+                    var span = request.Resolution.ToTimeSpan();
+                    var range = 0;
+
+                    while (utc < end)
+                    {
+                        var time = utc.ConvertFromUtc(sliceTimeZone);
+                        if (request.ExchangeHours.IsOpen(time, request.IncludeExtendedMarketHours))
```

Comment:
There's actually a method that does this in `Time.cs`:
```csharp
public static DateTime GetStartTimeForTradeBars(SecurityExchangeHours exchange, DateTime end, TimeSpan barSize, int barCount, bool extendedMarketHours)
{
    if (barSize <= TimeSpan.Zero)
    {
        throw new ArgumentException("barSize must be greater than TimeSpan.Zero", nameof(barSize));
    }

    var current = end.RoundDown(barSize);
    for (int i = 0; i < barCount;)
    {
        var previous = current;
        current = current - barSize;
        if (exchange.IsOpen(current, previous, extendedMarketHours))
        {
            i++;
        }
    }
    return current;
}
```

Reply:
`Time.GetStartTimeForTradeBars` is used to get the start time knowing the bar count, while we want the other way around. Maybe I could add `Time.GetBarsForPeriod` and use it instead.